AutomatticTracks/src/test/java/com/automattic/android/tracks/crashlogging/CrashLoggingTest.kt

```diff
@@ -1,65 +1,142 @@
 package com.automattic.android.tracks.crashlogging
 
+import android.app.Activity
 import com.automattic.android.tracks.TracksUser
 import com.automattic.android.tracks.fakes.FakeDataProvider
 import com.automattic.android.tracks.fakes.testUser
+import io.sentry.SentryEvent
+import io.sentry.SentryLevel
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.SoftAssertions
 import org.junit.Test
-import org.junit.runner.RunWith
-import org.mockito.Mockito.never
-import org.mockito.Mockito.spy
-import org.mockito.Mockito.times
-import org.mockito.Mockito.verify
-import org.robolectric.RobolectricTestRunner
-import org.robolectric.RuntimeEnvironment
-
-@RunWith(RobolectricTestRunner::class)
-class CrashLoggingTest {
 
-    lateinit var dataProvider: CrashLoggingDataProvider
-    var user: TracksUser? = null
+class CrashLoggingTest {
 
-    private fun initialize(currentUser: TracksUser? = null) {
+    val fakeProxy = FakeSentryErrorTrackerProxy()
```

Comment:
❓ What are advantages of using a fake object instead of a mock in this test suite? If we were using a mock, we could get rid of some boilerplate. I also believe that it'd be more consistent with other tests in our codebases. Wdyt?

Reply:
I like fakes for several reasons:
1. We don't have to have an additional dependency on `Mockito` and `Mockito inline`
2. Slightly faster test invocation (doesn't matter in such a small repo)
3. It's easier to adjust to any SUT changes (as there's an interface we have to satisfy, not mocks that have to be applied)

The cons is that I had to add a dummy interface we discussed above.

In the case of this repo, I don't have a favor in any direction. I moved to mocks in https://github.com/Automattic/Automattic-Tracks-Android/pull/51/commits/87027acd5715ed1cef55d3ff4a093e16fe6c0c9d as I like we could remove `SentryErrorTrackerProxy` interface.

Thanks for pointing out!

Reply:
Thanks for elaborating on this! 

I agree with your comments, but since we already depend on Mockito and it brings more features, I think moving to mocks was a good call. Thanks!